omniledger/service/proto.go

```diff
@@ -138,42 +138,34 @@ type Proof struct {
 
 // Instruction holds only one of Spawn, Invoke, or Delete
 type Instruction struct {
-	// InstanceID holds the id of the existing object that can spawn new objects.
-	// It is composed of the Darc-ID + a random value generated by OmniLedger.
+	// InstanceID is either the object that can spawn a new object, or a the object
+	// that will be invoked or deleted.
 	InstanceID InstanceID
-	// Nonce is monotonically increasing with regard to the darc in the instanceID
-	// and used to prevent replay attacks.
-	// The client has to track which is the current nonce of a darc-ID.
+	// Nonce is monotonically increasing with regard to the Darc controlling
+	// access to the instance. It is used to prevent replay attacks.
+	// The client has to track what the next nonce should be for a given instance.
 	Nonce Nonce
 	// Index and length prevent a leader from censoring specific instructions from
 	// a client and still keep the other instructions valid.
 	// Index is relative to the beginning of the clientTransaction.
 	Index int
 	// Length is the total number of instructions in this clientTransaction
 	Length int
-	// Spawn creates a new object
+	// Spawn creates a new object.
 	Spawn *Spawn
-	// Invoke calls a method of an existing object
+	// Invoke calls a method of an existing object.
 	Invoke *Invoke
-	// Delete removes the given object
+	// Delete removes the given object.
 	Delete *Delete
-	// Signatures that can be verified using the darc defined by the instanceID.
+	// Signatures that will be used while attempting to verify the Darc read request.
```

Comment:
`// Signatures that are verified using the darc referenced by InstanceID`

Reply:
I reworded yours as well because "which darc?"
```
// Signatures that are verified using the Darc controlling access to the instance.
```